openpype/hosts/blender/plugins/publish/extract_blender_scene_raw.py

```diff
@@ -0,0 +1,93 @@
+import os
+
+import bpy
+
+from openpype.pipeline import AVALON_CONTAINER_ID, publish
+from openpype.hosts.blender.api.workio import save_file
+
+
+class ExtractBlenderSceneRaw(publish.Extractor):
+    """Extract as Blender Scene (raw).
+
+    This will preserve all references, construction history, etc.
+    """
+
+    label = "Blender Scene (Raw)"
+    hosts = ["blender"]
+    families = ["blenderScene", "layout"]
+    scene_type = "blend"
+
+    def process(self, instance):
+        # Define extract output file path
+        dir_path = self.staging_dir(instance)
+        filename = "{0}.{1}".format(instance.name, self.scene_type)
+        path = os.path.join(dir_path, filename)
+
+        # We need to get all the data blocks for all the blender objects.
+        # The following set will contain all the data blocks from version
+        # 2.93 of Blender.
+        data_blocks = {
+            *bpy.data.actions,
+            *bpy.data.armatures,
+            *bpy.data.brushes,
+            *bpy.data.cache_files,
+            *bpy.data.cameras,
+            *bpy.data.collections,
+            *bpy.data.curves,
+            *bpy.data.fonts,
+            *bpy.data.grease_pencils,
+            *bpy.data.images,
+            *bpy.data.lattices,
+            *bpy.data.libraries,
+            *bpy.data.lightprobes,
+            *bpy.data.lights,
+            *bpy.data.linestyles,
+            *bpy.data.masks,
+            *bpy.data.materials,
+            *bpy.data.metaballs,
+            *bpy.data.meshes,
+            *bpy.data.movieclips,
+            *bpy.data.node_groups,
+            *bpy.data.objects,
+            *bpy.data.paint_curves,
+            *bpy.data.palettes,
+            *bpy.data.particles,
+            *bpy.data.scenes,
+            *bpy.data.screens,
+            *bpy.data.shape_keys,
+            *bpy.data.speakers,
+            *bpy.data.sounds,
+            *bpy.data.texts,
+            *bpy.data.textures,
+            *bpy.data.volumes,
+            *bpy.data.worlds
+        }
```

Comment:
Just wondering, wouldn't something dynamic like this work instead?

Like getting all members that are `bpy.types.struct` or even more specific are an [instance of `bpy.types.ID`](https://docs.blender.org/api/current/bpy.types.ID.html#bpy.types.ID) 

```python
data_blocks = set()
for attr in dir(bpy.data):
    data_block = getattr(bpy.data, attr)
    if isinstance(data_block, bpy.struct):
        data_blocks |= (*data_block)
```
Or:
```python
import inspect

def predicate(obj):
    return isinstance(obj, bpy.types.ID)

data_blocks = set()
for data_block in inspect.getmembers(byp.data, predicate):
    data_blocks |= (*data_block)
```

Please consider it pseudocode. :) I just wonder whether there's a way to get all the data dynamically?
I'm just wondering whether that might be more 'future-compatible'?

Reply:
I've actually tried to get the list of collection of data blocks with blender built-in functions, and didn't think of using `getattr` 😅 I'm going to commit this in a few minutes